src/RequestFactoryInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace GuzzleHttp;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Interface for generating HTTP requests.
+ */
+interface RequestFactoryInterface
+{
+
+    /**
+     * Applies any URI options to an existing URI to generate a new one.
+     *
+     * @param string $uri
+     * @param array  $config
+     *
+     * @return string
+     */
+    public function buildUri($uri, array $config);
```

Comment:
``createUri``?

Reply:
Sure, I just moved it here from `Client` and kept the name it had there